test/functional/sqlsrv/sqlsrv_ae_type_conversion_select.phpt

```diff
@@ -0,0 +1,302 @@
+                    }
+                    if ($convError[0][0] != '22018') {
+                        print_r($convError);
+                        fatalError("AE conversion failed with unexpected error message. i=$i, j=$j, v=$v\n");
+                    }
+                }
+            } else {
+                if ($conversionMatrix[$i][$j] == 'x') {
+                    fatalError("Conversion succeeded, should have failed. i=$i, j=$j, v=$v\n");
+                }
+                
+                if (AE\isDataEncrypted()) {
+                    $stmtAE = sqlsrv_query($conn, $selectQueryAE[$i][$j]);
+                    
+                    // Check every combination of statement value and conversion.
+                    // The last else if block covers the case where the select
+                    // query worked and the retrieved values are compared.
+                    if ($stmtAE == false and $conversionMatrixAE[$i][$j] == 'x') {
+                        $convError = sqlsrv_errors();
+                        if ($convError[0][0] != '22018') {
+                            print_r($convError);
+                            fatalError("AE conversion failed with unexpected error message. i=$i, j=$j, v=$v\n");
+                        }
+                    } elseif ($stmtAE == false and $conversionMatrixAE[$i][$j] == 'y') {
+                        print_r(sqlsrv_errors());
+                        fatalError("AE conversion failed, should have succeeded. i=$i, j=$j, v=$v\n");
+                    } elseif ($stmtAE != false and $conversionMatrixAE[$i][$j] == 'x') {
+                        fatalError("AE conversion succeeded, should have failed. i=$i, j=$j, v=$v\n");
+                    } elseif ($stmtAE != false and $conversionMatrixAE[$i][$j] == 'y') {
+                        $row = sqlsrv_fetch_array($stmt, SQLSRV_FETCH_NUMERIC);
+                        $rowAE = sqlsrv_fetch_array($stmtAE, SQLSRV_FETCH_NUMERIC);
+                        
+                        // rtrim strips whitespace from the end of the string, which 
+                        // takes care of a bug where some conversions lead to extraneous
+                        // whitespace padding the end of the string
+                        if (is_string($row[0])) {
+                            $row[0] = rtrim($row[0]);
+                            $rowAE[0] = rtrim($rowAE[0]);
+                        }
+                        
+                        if ($row[0] != $rowAE[0]) {
+                            echo "Values do not match! i=$i, j=$j, v=$v\n";
+                            print_r($row[0]);
+                            print_r($rowAE[0]);
+                            fatalError("Test failed, values do not match\n");
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    $deleteQuery = "DELETE FROM $tableName";
+    $stmt = sqlsrv_query($conn, $deleteQuery);
```

Comment:
Please use `TRUNCATE $tableName` instead
From DOCS: "Removes all rows from a table or specified partitions of a table, without logging the individual row deletions. TRUNCATE TABLE is similar to the DELETE statement with no WHERE clause; however, TRUNCATE TABLE is faster and uses fewer system and transaction log resources."